[MRG+1] Fix: Replace pylab with matplotlib.pyplot #6754

ztultrebor opened the pull request:
#6754 

<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#Contributing-Pull-Requests
-->
#### Reference Issue

<!-- Example: Fixes #1234 -->
#### What does this implement/fix? Explain your changes.
- multiple instances of occurrences of pylab replaced with matplotlib.pyplot
- references to pl ( as in import pylab as pl) replaced 
  with plt (as in import matplotlib.pyplot as plt)
#### Any other comments?
- one file still contains references to pylab: doc/sphinxext/gen_rst.py
- I'd need some time to figure out what needs to be done; closing this for now

<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

ztultrebor committed:
Fix: Replace pylab with matplotlib.pyplot #6754

 - one instance of 22 occurrences of pylab replaced with matplotlib.pyplot
 - bench_glm.py now free of pylab references
 - code executes properly

agramfort commented:
there is only one file changed in this PR so far

ztultrebor committed:
Fix: Replace pylab with matplotlib.pyplot #6754

- one instance of 21 remaining occurrences of pylab replaced with
    matplotlib.pyplot
- bench_glmnet.py now free of pylab references
- code does not execute for extraneous reason: ImportError: No module named
    glmnet.elastic_net

ztultrebor committed:
Fix: Replace pylab with matplotlib.pyplot #6754

- one instance of 19 occurrences of pylab replaced with matplotlib.pyplot
- bench_lasso.py now free of pylab references
- code executes properly

ztultrebor committed:
Fix: Replace pylab with matplotlib.pyplot #6754

- one instance of 18 occurrences of pylab replaced with matplotlib.pyplot
- bench_plot_neighbors.py now free of pylab references
- code executes properly

ztultrebor committed:
Fix: Replace pylab with matplotlib.pyplot #6754

- one instance of 17 occurrences of pylab replaced with matplotlib.pyplot
- bench_plot_omp_lars.py now free of pylab references
- code does not execute for extraneous reasons:
    - File "bench_plot_omp_lars.py", line 111, in <module>
    - ax = fig.add_subplot(1, 2, i)
    - ValueError: num must be 1 <= num <= 2, not 0
    - line 111 should probably be ax = fig.add_subplot(1, 2, i+1)

ztultrebor committed:
Fix: Replace pylab with matplotlib.pyplot #6754

- bench_plot_parallel_pairwise.py now free of pylab references
- code executes properly

ztultrebor committed:
Fix: Replace pylab with matplotlib.pyplot #6754

- bench_plot_ward.py now free of pylab references
- code executes properly

ztultrebor committed:
Fix: Replace pylab with matplotlib.pyplot #6754

- bench_sgd_regression.py now free of pylab references
- code executes properly

ztultrebor committed:
Fix: Replace pylab with matplotlib.pyplot #6754

- bench_tree.py now free of pylab references
- code executes properly

agramfort commented on benchmarks/bench_glm.py:
```diff
@@ -12,7 +12,7 @@
 if __name__ == '__main__':
 
-    import pylab as pl
+    import matplotlib.pyplot as pl
```
standard import name is plt

please change all occurrences.

thanks

ztultrebor committed:
refactor: Replace pl with plt

- replace instances of pl (as on import pylab as pl)
  with plt (as in import matplotlib.pyplot as plt)
- bench_glm.py clean

ztultrebor committed:
refactor: Replace pl with plt

- replace instances of pl (as on import pylab as pl)
  with plt (as in import matplotlib.pyplot as plt)
- bench_glm.py clean of pl
- code does not execute for extraneous reasons

ztultrebor committed:
refactor: Replace pl with plt

- replace instances of pl (as on import pylab as pl)
  with plt (as in import matplotlib.pyplot as plt)
- bench_lasso.py clean of pl
- code executes properly

ztultrebor committed:
refactor: Replace pl with plt

- replace instances of pl (as on import pylab as pl)
  with plt (as in import matplotlib.pyplot as plt)
- bench_plot_neighbors.py clean of pl
- code executes properly

ztultrebor committed:
refactor: Replace pl with plt

- replace instances of pl (as on import pylab as pl)
  with plt (as in import matplotlib.pyplot as plt)
- bench_plot_omp_lars.py clean of pl
- code does not execute for extraneous reasons

ztultrebor committed:
fix: Fix bug that prevented graphs from displaying

ztultrebor committed:
refactor: Replace pl with plt

- replace instances of pl (as on import pylab as pl)
  with plt (as in import matplotlib.pyplot as plt)
- bench_plot_parallel_pairwise.py clean of pl
- code executes properly

ztultrebor committed:
refactor: Replace pl with plt

- replace instances of pl (as on import pylab as pl)
  with plt (as in import matplotlib.pyplot as plt)
- bench_plot_ward.py clean of pl
- code executes properly

ztultrebor committed:
refactor: Replace pl with plt

- replace instances of pl (as on import pylab as pl)
  with plt (as in import matplotlib.pyplot as plt)
- bench_sgd_regression.py clean of pl
- code executes properly

ztultrebor committed:
refactor: Replace pl with plt

- replace instances of pl (as on import pylab as pl)
  with plt (as in import matplotlib.pyplot as plt)
- bench_tree.py clean of pl
- code executes properly

ztultrebor committed:
docs: removed pylab references from comments

ztultrebor committed:
docs: removed all pylab references

- replaced with matplotlib.pyplot
- pl --> plt

ztultrebor committed:
docs: removed pylab references from comments

- replaced with matplotlib.pyplot
- pl --> plt

ztultrebor committed:
docs: removed all pylab references

- replaced with matplotlib.pyplot
- pl --> plt

ogrisel commented on doc/tutorial/statistical_inference/supervised_learning.rst:
```diff
@@ -212,15 +212,15 @@ induces high variance:
 
-    >>> import pylab as pl # doctest: +SKIP
-    >>> pl.figure() # doctest: +SKIP
+    >>> import matplotlib as plt # doctest: +SKIP
```
typo: `matplotlib.pyplot`

ogrisel commented:
Thanks @ztultrebor. Are you done with this PR? If so please change the title from [WIP] to [MRG].

+1 for merge on my side once https://github.com/scikit-learn/scikit-learn/pull/6762#discussion_r62462638 is taken care of.

ztultrebor commented:
I've still got a few more files to clean up. Once those are taken care of I'll be certain to update the pull title appropriately

ztultrebor committed:
refactor: Replace pl with plt

- replace instances of pl (as on import pylab as pl)
  with plt (as in import matplotlib.pyplot as plt)
- mlcomp_sparse_document_classification.py clean of pl

ztultrebor committed:
refactor: Replace pl with plt

- replace instances of pl (as on import pylab as pl)
  with plt (as in import matplotlib.pyplot as plt)
- plot_gpr_noisy_targets.py clean of pl
- code does not execute for extraneous reasons
  - File "examples/gaussian_process/plot_gpr_noisy_targets.py", line 31, in
      <module>
  - from sklearn.gaussian_process import GaussianProcessRegressor
  - ImportError: cannot import name GaussianProcessRegressor

ztultrebor committed:
refactor: Replace pl with plt

- replace instances of pl (as on import pylab as pl)
  with plt (as in import matplotlib.pyplot as plt)
- plot_gpc_isoprobability.py clean of pl
- code does not execute for extraneous reasons
  - File "examples/gaussian_process/plot_gpc_isoprobability.py", line 24, in
      <module>
  - from sklearn.gaussian_process import GaussianProcessClassifier
  - ImportError: cannot import name GaussianProcessClassifier

ztultrebor committed:
docs: removed all pylab references

- replaced with matplotlib.pyplot
- pl --> plt

ztultrebor committed:
docs: removed all pylab references

- replaced with matplotlib.pyplot

ztultrebor committed:
refactor: Replace pl with plt

- replace instances of pl (as on import pylab as pl)
  with plt (as in import matplotlib.pyplot as plt)
- plot_sparse_coding.py clean of pl
- code executes properly

ztultrebor committed:
docs: removed all pylab references

- replaced with matplotlib.pyplot

ztultrebor committed:
docs: removed all pylab references

- replaced with matplotlib.pyplot

TomDLT commented on benchmarks/bench_plot_neighbors.py:
```diff
@@ -131,21 +131,21 @@ def barplot_neighbors(Nrange=2 ** np.arange(1, 11),
 
-            c_bar = pl.bar(xvals, build_time[alg] - bottom,
+            c_bar = plt.bar(xvals, build_time[alg] - bottom,
                            width, bottom, color='r')
```
you need to change the indentation here and there, since you added a character

TomDLT commented on doc/tutorial/text_analytics/skeletons/exercise_01_language_train_model.py:
```diff
@@ -46,9 +46,9 @@
-#pl.matshow(cm, cmap=pl.cm.jet)
-#pl.show()
+#import matplotlib.pyplot as plt
+#t.matshow(cm, cmap=plt.cm.jet)
```
typo `t` -> `plt`

ztultrebor committed:
style: Indent properly

ztultrebor committed:
style: indent properly

ztultrebor committed:
style: Indent properly

ztultrebor committed:
docs: Add missing .pyplot

ztultrebor committed:
docs: Fix typo

ztultrebor committed:
style: Indent properly

TomDLT commented:
Thank you @ztultrebor !
